app/database/operator/server_data_operator/handlers/post.ts

```diff
@@ -409,51 +439,53 @@ const PostHandler = <TBase extends Constructor<ServerDataOperatorBase>>(supercla
         ).fetch()) as PostsInChannelModel[];
 
         // chunk length 0; then it's a new chunk to be added to the PostsInChannel table
-        if (chunks.length === 0) {
+        if (!chunks.length) {
             return this._createPostsInChannelRecord(channelId, earliest, latest, prepareRecordsOnly);
         }
 
         let targetChunk: PostsInChannelModel|undefined;
 
         for (const chunk of chunks) {
-            // find if we should plug the chain before
-            if (firstPost.create_at >= chunk.earliest || latest <= chunk.latest) {
```

Comment:
This was making non intersecting chunks to be considered part of the target, potentially creating gaps.